### PR TITLE
Feature database migration

### PR DESCRIPTION
**Pull request by nmonga91**
Implemented database migrations to automate database schema changes. Also added alembic to the list of requirements. Updated documentation to include instructions on how to run migrations.

### CODE REVIEW

File: dataactcore/db_cleanup.sh [hunk 1]
```diff
@@ -0,0 +1,5 @@
+psql -U postgres -d nipun_test -c 'drop schema public cascade; create schema public;'
```
Comment (jworcestBAH) [inline dataactcore/db_cleanup.sh]: What is this line doing, and should it be part of this script?  Also, could we add a comment at the top of this script saying what it does?

File: dataactcore/db_cleanup.sh [hunk 2]
```diff
@@ -0,0 +1,5 @@
...
+psql -U postgres -d error_data -c 'drop schema public cascade; create schema public;'
+psql -U postgres -d job_tracker -c 'drop schema public cascade; create schema public;'
+psql -U postgres -d user_manager -c 'drop schema public cascade; create schema public;'
+psql -U postgres -d validation -c 'drop schema public cascade; create schema public;'
```
Comment (jworcestBAH) [inline dataactcore/db_cleanup.sh]: Should the core version of this have a reference to the validation DB?  Is there a version of this in validator that we could move this to?

File: dataactcore/db_cleanup.sh (hunk 1 above)
Comment (nmonga91) [inline dataactcore/db_cleanup.sh]: @jworcestBAH That script shouldn't be included at all. I'll remove that in the next commit. It was just there for my cleanup purposes.

File: dataactcore/db_cleanup.sh (hunk 2 above)
Comment (nmonga91) [inline dataactcore/db_cleanup.sh]: @jworcestBAH This script is being omitted.

File: dataactcore/migrations/env.py [hunk 3]
```diff
@@ -0,0 +1,152 @@
...
+#}
+
+target_metadata = {
+    'error_data':errorModels.Base.metadata,
+    'job_tracker':jobModels.Base.metadata,
+    'user_manager':userModel.Base.metadata
+}
+
+# Set up database URLs based on credentials file
+with open('credentials/dbCred.json') as data_file:
```
Comment (jworcestBAH) [inline dataactcore/migrations/env.py]: Can we refactor this to use BaseInterface.getCredDict?  Could have the BaseInterface credFileName variable point to dbCred.json as a default value.

**Pull request merged**